### ISSUE TITLE
`function-paren-newline`: add a case for function taking a callback as the single argument

### ISSUE THREAD

**Issue by Ragnar-Oock** [eslint-stylistic/eslint-stylistic#290]
### Clear and concise description of the problem

Take the following code : 
```ts
const array = [ 'bob', 'billy' ];
array.find(arrayItem => 
  arrayItem.startsWith('bob')
);
```

There is no configuration of `@stylistic/function-paren-newline` that would keep this code formatted that way, with a line break after the arrow (with or without curlies) and one before the last parenthesis :
- 'consistent' or 'always' expects a break before the start of the predicate to keep the last paren on its line
- 'multiline' and 'multiline-arguments' doesn't apply to arrow function bodies

### Suggested solution

A new option ('callback-only' ?) could be added that allows/expect a line break before the closing paren of a function taking a single callback argument.

### Alternative

I tried all the other options of `function-paren-newline` but the only way to allow this style is disabling the rule (which I don't want to do)

### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [ ] If this feature request is accepted, I am willing to submit a PR to fix this issue

**Comment by jimmy-zhening-luo** [comment]
+1, and a re-statement (correct me if this is wrong):

There exists a condition to force a new line if there are linebreaks between arguments.

I would like a condition to force a new line if there are linebreaks **within** an argument.

Examples:

1. One-argument function call, where the argument is an arrow-paren split onto two lines by rule `arrow-parens`
2. One-argument function call, where the argument is a set of logical conditions split across many lines (e.g. a`\n`&& b)
3. One-argument function call, where the argument is a set of chained operators (e.g. a.b`\n`.c`\n`.d, a`\n`+ b`\n`+ c)

**Comment by Ragnar-Oock** [comment]
This issue has been marked as complete and published in 5.5.0, however the snippet from the original post is still flagged as erroneous. There is still no option for `function-paren-newline` to allow no new line before the first param and a new line before the closing parens.
